crates/evm/src/execute.rs

```diff
@@ -2,12 +2,73 @@
 
 use reth_primitives::{BlockNumber, BlockWithSenders, Receipt, Receipts, Request, Requests, U256};
 use reth_prune_types::PruneModes;
-use revm::db::BundleState;
-use revm_primitives::db::Database;
+use revm::{db::BundleState, Context, DatabaseCommit, Evm};
+use revm_primitives::{db::Database, EVMError, ResultAndState};
 
 pub use reth_execution_errors::{BlockExecutionError, BlockValidationError};
 pub use reth_storage_errors::provider::ProviderError;
 
+/// An EVM executor trait that executes an input (e.g. transaction and height) and produces an output
+/// (e.g. state changes).
+///
+/// The executor cannot commit the state changes to the database directly, see [`EvmCommitter`].
+pub trait EvmExecutor {
```

Comment:
I see, this essentially mimics the `revm::Evm`, right?

Reply:
Yes, keeping just the minimum API to be able to transact